homeassistant/components/climate/tado.py

```diff
@@ -160,26 +203,40 @@ def current_temperature(self):
     @property
     def current_operation(self):
         """Return current readable operation mode."""
-        if self._cooling:
-            return "Cooling"
+        if self._current_operation in OPERATION_MANUAL:
+            if self._mode == CONST_MODE_COOL:
+                return OPERATION_MANUAL_COOL.get(self._current_operation)
+            if self._device_type == "HOT_WATER" and \
+                    not self._is_manual_temperature:
+                return OPERATION_MANUAL_HOT_WATER_ON.get(
+                    self._current_operation)
+            return OPERATION_MANUAL.get(self._current_operation)
         return OPERATION_LIST.get(self._current_operation)
 
     @property
     def operation_list(self):
         """Return the list of available operation modes (readable)."""
-        return list(OPERATION_LIST.values())
+        operations = list(OPERATION_LIST.values())
```

Comment:
This is all wrong. Operation modes should be limited to the states as provided by the Home Assistant Climate component: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/climate/__init__.py#L43-L50

No other modes are allowed because it breaks the integration with Alexa, Google Assistant etc.

Reply:
Okay, so let's close this PR then?

Reply:
??? so you want to remove the tado integration from hass!?

Reply:
@balloob
STATE_HEAT = 'heat'
STATE_COOL = 'cool'
STATE_IDLE = 'idle'
STATE_AUTO = 'auto'
STATE_MANUAL = 'manual'
STATE_DRY = 'dry'
STATE_FAN_ONLY = 'fan_only'
STATE_ECO = 'eco'

Tado do not use those singular states, if you want to revamp Hass to cater tado it need both heat and cool states

Heat_auto
Heat_manual
Heat_timer
Heat_off

Cool_auto
Cool_manual
Cool_timer
Cool_off

Due to the fact tado can handle both hot and cold not just hot in many thermostat

Reply:
Nobody will turn on the heating and the cooling at the same time so that can easily be represented in a single operation mode.

I am not discussing this, I don't have time for that. PRs that do not follow our states are not accepted, period. 

So shall I close this PR?

Reply:
I wouldn't discuss this, if it would be a new component. But this is only a change to an existing one!

We change the PR to WIP and look if we can integrate the default states. I would guess that `timer` and `manual` are nearly the same in tado and `auto` is the tado schedule mode. So we may need a translation for hass states to tado states.

Reply:
I suppose we could reuse it like :

STATE_AUTO = 'auto' = tado_mode
STATE_MANUAL = 'manual' = manual
STATE_ECO = 'eco' = smart schedule

Would make it more confusing for the users but we keep all of the current functionality and I suppose we would Follow homeassistant states

Reply:
we can leave it nearly the same for the user, I suppose! We already did a "translation" for the UI and the Tado States. So now, we may limit the states to those in the climate devices and map the UI to those states. After that we map the climate devices steps to tado states (this should best be done inside the method, where we actually communicate with tado/PyTado)

Reply:
And "timer" is like
-> "manual" + switch back to "tado schedule" after a timespan.
  or
-> "manual" + switch back to "tado schedule" when tado would change it next.

whereas "manual" is like
-> "manual" + never switch back

and "auto"
-> "tado schedule mode"

So we may have to do more checks, what is the current tado state and set tado states accordingly. Currently we are doing this a bit too simple, because we just switch to "manual" and set the temperature, we do not change the tado "manual" properties like the time when it will switch back or if it should wait until tado switches it back manually.

Reply:
Currently manual just means that the user needs manually stop the overlay. Since manual is an overlay right? And there is no option to add a time for the timer from the UI which basically renders it useless unless we hardcode this time?

Sorry I think I kind of miss your point/Don't understand?

Reply:
I guess my point was that we don’t really need to change the behavior 😜 We just need to rename the tado overlays to the modes which are used within homeassistant right?